source/common/upstream/health_discovery_service.cc

```diff
@@ -273,18 +272,18 @@ HdsCluster::HdsCluster(Server::Admin& admin, Runtime::Loader& runtime,
                        Ssl::ContextManager& ssl_context_manager, bool added_via_api,
                        ClusterInfoFactory& info_factory, ClusterManager& cm,
                        const LocalInfo::LocalInfo& local_info, Event::Dispatcher& dispatcher,
-                       Random::RandomGenerator& random, Singleton::Manager& singleton_manager,
-                       ThreadLocal::SlotAllocator& tls,
+                       Singleton::Manager& singleton_manager, ThreadLocal::SlotAllocator& tls,
                        ProtobufMessage::ValidationVisitor& validation_visitor, Api::Api& api)
     : runtime_(runtime), cluster_(std::move(cluster)), bind_config_(bind_config), stats_(stats),
       ssl_context_manager_(ssl_context_manager), added_via_api_(added_via_api),
       initial_hosts_(new HostVector()), validation_visitor_(validation_visitor) {
   ENVOY_LOG(debug, "Creating an HdsCluster");
   priority_set_.getOrCreateHostSet(0);
 
-  info_ = info_factory.createClusterInfo(
-      {admin, runtime_, cluster_, bind_config_, stats_, ssl_context_manager_, added_via_api_, cm,
-       local_info, dispatcher, random, singleton_manager, tls, validation_visitor, api});
+  info_ = info_factory.createClusterInfo({admin, runtime_, cluster_, bind_config_, stats_,
+                                          ssl_context_manager_, added_via_api_, cm, local_info,
+                                          dispatcher, api.randomGenerator(), singleton_manager, tls,
```

Comment:
you are passing api below, so maybe you don't need to pass the RNG separately?

Reply:
I'm splitting this refactor across a few PRs, gonna address this one in the next PR.

Reply:
The suggestion here is to make *less* changes in this PR, right?

Reply:
oh, never mind. I mis-read the 'before' side. What you have is fine for now given there is a follow-up planned.

Reply:
Fixed, I didn't realize the params object could be modified without changing a bunch of call sites.